src/hooks/useUser.ts

```diff
@@ -3,26 +3,26 @@ import { fetchUserProfile, updateUserProfile } from '@/api/userApi';
 import { QUERY_KEYS } from '@/utils/queryKeys';
 import type { UserProfile } from '@/types/User';
 
-export const useUserProfile = () => {
+export const useUserProfile = (userId: number) => {
   return useQuery<UserProfile>({
-    queryKey: QUERY_KEYS.user.profile(),
-    queryFn: fetchUserProfile,
-    staleTime: 1000 * 60 * 5, 
-    retry: 1, 
+    queryKey: QUERY_KEYS.user.profile(userId),
```

Comment:
타입 가드를 붙이는 건 어떨까 제안해봅니다!!


```suggestion
import { USER_CONSTANTS } from '@/constants/user';

    queryKey: QUERY_KEYS.user.profile(userId ?? USER_CONSTANTS.INVALID_USER_ID),
```

Reply:
넵 타입가드 처리했습니다~